crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -753,6 +775,9 @@ fn finalize_block<Block, Backend, Client>(
 /// processing, which is necessary for ensuring that when the next block is imported, inherents will
 /// contain segment header of newly archived block (must happen exactly in the next block).
 ///
+/// When a block with object mappings is archived, notification ([`SubspaceLink::archived_mapping_notification_stream`])
+/// will be sent. If any mappings spill over to the next segment, they will be sent when the next block is archived.
```

Comment:
I would remove this second sentence. You're explaining implementation details while what object mappings are produced when is already out of user's control anyway. I'd keep the order not defined to not impose additional contracts on the implementation. Also I hope what is described in this sentence will not be true eventually anyway.